test/autotest.py

```diff
@@ -506,6 +506,8 @@ def testKill():
   def testCheckpoint():
     #start checkpoint
     coordinatorCmd(CKPT_CMD)
+    #wait and see if process will die shortly after checkpoint
+    sleep(S*SLOW)
```

Comment:
The pseudo-code you propose would require moving CHECK(...) outside of testCheckpoint().  Currently, CHECK(...) is part of doing a testCheckpoint(), which fits better with the name testCheckpoint().

It's true that the WAITFOR() inside testCheckpoint() will timeout if it doesn't see the correct number of checkpoint files create.  So how about if I move sleep() until after the WAITFOR():
```
def testCheckpoint():
    coordinatorCmd(CKPT_CMD)
    WAITFOR(...getNumCkptFiles(ckptDir)>0 and...)
    sleep(S*SLOW)
    CHECK(doesStatusSatisfy(getStatus(), status),
```
This keeps close to the original logic, and preserves the meaning behind the name `testCheckpoint()`.